sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/ContainerAsyncClient.java

```diff
@@ -300,7 +304,7 @@ public URL getContainerUrl() {
         if (!validateNoEtag(accessConditions.modifiedAccessConditions())) {
             // Throwing is preferred to Single.error because this will error out immediately instead of waiting until
             // subscription.
-            throw new UnsupportedOperationException("ETag access conditions are not supported for this API.");
+            logger.logAndThrow(new UnsupportedOperationException(ETAG_NOT_ALLOWED_MESSAGE));
```

Comment:
It is not really needed to add a new message field. No need to be consistent. If you think the entire block is duplicate, it is much better to have a helper method.